### DISCUSSION TITLE
Hard forcing recognition of a word

### DISCUSSION THREAD

**Discussion by chuddy1** [openai/whisper#222] [Q&A] [answered]
Hi thank you for this program. I am trying to do phrase detection within a noisy environment. I am trying to detect one-word phrases such as "corner" and "behind". The program detects the sound of the word, but does not consistently transcribe the word (i know this may be due to the noisiness of the environment)

I would like to know if there was a way that I could force the software to pick certain one word phrases if they are among the best match while detecting (i.e., if "corner" or "behind" are included in best match, then it should return any of those words). I know this would cause some false positives when detecting those certain phrases, but in my case it does not matter as much.

Thank you!

**Comment by jianfch** [comment, marked as answer]
You can have it return predicted probabilities of those tokens that make up the phrase/words you want along decoded results.

**Reply by chuddy1** [reply]
Thank you for your reply. I had a similar idea, but how do I get the predicted probabilities for that token?

**Reply by jianfch** [reply]
the [logits](https://github.com/openai/whisper/blob/0b1ba3d46ebf7fe6f953acfd8cad62a4f851b49f/whisper/decoding.py#L595) are the predicted probabilities 
```python
logits_for_corner_behind = logits[:, self.tokenizer.encode('corner behind')]
```

**Reply by nmrugg** [reply]
Is there any way to get a list of possible transcriptions for a word or phrase?
